pom.xml

```diff
@@ -532,6 +551,25 @@
               </rules>
             </configuration>
           </execution>
+          <execution>
+            <id>no-snapshots-in-release</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- 
+                  this causes fails the incrementals IT to fail as the parent is a snapshot and the incremental is a release
+                  so this is a separate execution so that it can be skipped in that specific IT
+                  -->
+                <requireReleaseDeps>
+                  <message>No SNAPSHOT versions are allowed for releases</message>
+                  <onlyWhenRelease>true</onlyWhenRelease>
+                </requireReleaseDeps>
```

Comment:
there have been the odd occaisoin where the maven-release-plugin has created a release with snapshots.
this also prevents incremental from being pushed that use snapshots (they would be non-deterministic when used which would defeat the point of a incremental revision).

the downside is that it makes it harder to actually test a new version of this pom when used in an incremental version. (the profile would need to be skipped to deploy) (because this pom itself is not published as an incremental)

//cc @jglick for any comments on this

Reply:
Not sure if there is any clean solution to this. I recall suppressing this rule before for that exact reason.

Reply:
added `failWhenParentIsSnapshot` which will allow easier testing of SNAPSHOT `plugin-pom`s at the expense of reproducability (however given that we did not check before this may be the lesser of 2 evils currently)